client/src/components/events-components/EventModal.tsx

```diff
@@ -0,0 +1,63 @@
+import type { EventWithAllData } from "@/utils/interfaces";
+import {
+  Dialog,
+  DialogContent,
+  DialogDescription,
+  DialogHeader,
+  DialogTitle,
+} from "../ui/dialog";
+import { useEffect, useState } from "react";
+import { getEvent } from "@/utils/eventService";
+
+interface EventModalProps {
+  open: boolean;
+  onOpenChange: (open: boolean) => void;
+  eventId: number;
+}
+const EventModal = ({ open, onOpenChange, eventId }: EventModalProps) => {
+  const [isEventLoading, setIsEventLoading] = useState(true);
+  const [event, setEvent] = useState<EventWithAllData | null>(null);
+  useEffect(() => {
+    const fetchEvent = async () => {
+      const fetchedEvent = await getEvent(eventId);
+      setEvent(fetchedEvent);
+      setIsEventLoading(false);
+    };
+    fetchEvent();
+  }, []);
+
+  if (isEventLoading) {
+    return <></>;
+  }
+  return (
+    <Dialog open={open} onOpenChange={onOpenChange}>
```

Comment:
is this the only modal in the application? otherwise you should consider adding that in the root that any components can call through a context and hook (e.g. `useGlobalDialog`)

this can prevent having multiple `<Dialog>` in the HTML body